pkg/operator/configobservation/apiserver/observe_termination_duration.go

```diff
@@ -0,0 +1,131 @@
+package apiserver
+
+import (
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var shutdownDelayDurationPath = []string{"apiServerArguments", "shutdown-delay-duration"}
+
+var watchTerminationDurationPath = []string{"watchTerminationDuration"}
+
+// ObserveShutdownDelayDuration allows for overwriting shutdown-delay-duration value.
+// It exists because the time needed for an LB to notice and remove unhealthy instances might vary by platform.
+func ObserveShutdownDelayDuration(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains shutdown-delay-duration field.
+		ret = configobserver.Pruned(ret, shutdownDelayDurationPath)
+	}()
+	var currentShutdownDelayDuration string
+
+	// read the current value
+	currentShutdownDelaySlice, _, err := unstructured.NestedStringSlice(existingConfig, shutdownDelayDurationPath...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("unable to extract shutdown delay duration from the existing config: %v", err))
+	}
+
+	if len(currentShutdownDelaySlice) > 0 {
+		currentShutdownDelayDuration = currentShutdownDelaySlice[0]
+	}
+
+	// read the observed value
+	var observedShutdownDelayDuration string
+	listers := genericListers.(configobservation.Listers)
+	infra, err := listers.InfrastructureLister().Get("cluster")
+	if err != nil && !apierrors.IsNotFound(err) {
+		// without the infrastructure object we are not able to determine the type of platform we are running on
+		return existingConfig, append(errs, err)
+	}
+	switch infra.Spec.PlatformSpec.Type {
+	case configv1.AWSPlatformType:
+		// AWS has a known issue: https://bugzilla.redhat.com/show_bug.cgi?id=1943804
+		// We need to extend the shutdown-delay-duration so that an NLB has a chance to notice and remove unhealthy instance.
+		// Once the mentioned issue is resolved this code must be removed and default values applied
+		observedShutdownDelayDuration = "210s"
+	default:
+		observedShutdownDelayDuration = currentShutdownDelayDuration
```

Comment:
I expected a constant here, not the old value.

Reply:
the old value contains the default value set in https://github.com/openshift/cluster-kube-apiserver-operator/blob/6daaae3f1a084e3ab637195ba4583b365e0eac37/bindata/v4.1.0/config/defaultconfig.yaml#L148

Reply:
we definitely don't want to run a server with this value set to 0

Reply:
Then this should mean not to set it at all, shouldn't it? Take the old value I think is wrong.

Reply:
In this case, the old value is actually no value. I can return early to make it more readable.